### DISCUSSION TITLE
BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES

### DISCUSSION THREAD

**Discussion by jfpe37** [firemodels/fds#13005] [General]
Good Morning
I received the attached error "**BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES**", After 391 seconds.
(Blowers start to work in 65.9 seconds)
I am attaching the FDS file after I removed the objects of the structure due to the need to maintain confidentiality.
I run on a computer with 40 cores and 1 terabyte of memory. 
(Attaches the data).
I have another run on this computer (a different simulation) that didn't crash.
I allso run the simulation using the command "mpiexec -np 4 fds LG103_304.fds", and actually want to use 4 processors when I have 4 mesh
Is the command correct?
I would appreciate your help in understanding the error.
thanks
Mor
![Capture2](https://github.com/firemodels/fds/assets/118098213/4a15e86f-f134-4ae1-822c-6132af425269)
![Capture1](https://github.com/firemodels/fds/assets/118098213/50c63b77-135e-4cc6-a519-92d6c95f3fcc)
[fds 304.txt](https://github.com/firemodels/fds/files/15480412/fds.304.txt)

**Comment by Er9y714** [comment]
Is this running on the latest FDS version 6.9.1?
Does this issue repeat when you tried re-runing? Or is this the first and only case this happened? I

**Reply by jfpe37** [reply]
Hi
I am running on FDS-6.8.0
This is not the first time this has happened.
It also happened to me on another run of the same project in a different fire scenario.
I just ran the simulation one more time - by doing a RESTART
When I changed the run command to the following form:
**fds_local -p 4 -o 2 LG103_304.fds**
Does it matter?
The run was 5 seconds past the time when it got stuck last time.

Do you have any insights on the matter?
I continue to follow and hope it doesn't get stuck again
thank you
Mor

**Reply by Er9y714** [reply]
-p 4 -o 2 might run differently than -p 4 only. I would not change the running command while restarting. 

Note that the support is provided for the latest software version. Because there are lots of fixes being done everyday. Furthermore it is not productive to focus on old versions since the issue might have been fixed already in the latest version. If you need support, it is best to run with the latest version and if you still have an issue, submit it here so the developers can try to reproduce.

**Comment by mcgratta** [comment]
With the latest version of FDS, I get these warnings and error:
```
WARNING: SPEC POLYURETHANE_REAC is not in the table of pre-defined species. Any unassigned SPEC variables in the input were assigned the properties of nitrogen.
WARNING: Droplet heat transfer is not predicted when a droplet is on a SURF with a specified ADIABATIC, NET_HEAT_FLUX, or CONVECTIVE_HEAT_FLUX.

ERROR(820): VENT intake SMOKE GF-075 has no solid backing or an orientation index (IOR) is needed. (MPI Process: 0, CHID: LG103_304)

```
The error that stops the job has to do with a vent being assigned to an obstruction that is either small or thin or otherwise not appropriate.

**Reply by jfpe37** [reply]
Thanks
I will check the above vent.
How can I see the errors? Does it only exist in the newest version?

**Comment by mcgratta** [comment]
Yes, use the latest version. If you still have trouble, we can help you. Also, your meshes are very large. I suggest you use smaller meshes and more cores to run the job more quickly. I cannot help you if the job crashes after a week.

**Reply by jfpe37** [reply]
Thank you for the answer.
What is the recommendation for the amount of cells in each MESH
How many processors should be used for this type of running to maximize running speed?
300 cells for one MESH?
Running 8 MESH on 8 cores?
Thanks

**Reply by drjfloyd** [reply]
In general the more MPI processes you use, the faster FDS will run as long as your machine can support that number of MPI processes on the hardware.  For 3D geometries, I don't generally put less than 10,000 - 20,000 cells per mesh when using multiple meshes. If you are using one of the alternate pressure solvers you do want to keep meshes smaller (see the User's Guide). 

I recommend you turn off virtualization for the machine. You will likely have a significant performance hit if windows tries to assign multiple FDS processes to the same physical core.

The best way to determine how many processes you can use for FDS is to benchmark your PC.  You can run the scaling tests we document in the User's Guide (https://github.com/firemodels/fds/tree/master/Validation/MPI_Scaling_Tests/FDS_Input_Files).  If you don't want to take that time, I'd recommend limiting FDS to 38 cores and leaving a couple for Windows processes.  For this input that would put you aroud 80,000 cells per mesh.
